tests/test_dataset_implementations/test_dataset_implementations.json

```diff
@@ -0,0 +1,378 @@
+{
+    "test_datasets": {
+        "d1": {
+            "data": {
+                "x": [2.0, 5.0],
+                "y": [5, 5],
+                "z": [0, 10],
+                "n": [0, null],
+                "b": [true, false]
+            },
+            "schemas": {
+                "SparkDFDataset": {
+                  "x": "float",
+                  "y": "int",
+                  "z": "int",
+                  "n": "int",
+                  "b": "bool"
+                }
+            }
+        },
+        "d2": {
+            "data": {
+                "a": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+                "b": ["a", "b", "b", "c", "c", "c", "d", "d", "d", "d"],
+                "c": ["a", "b", "b", "c", "c", "c", "d", null, null, null],
+                "d": ["a", "b", "c", "d", "e", "f", "g", null, null, null]
+            },
+            "schemas": {
+                "SparkDFDataset": {
+                  "a": "int",
+                  "b": "string",
+                  "c": "string",
+                  "d": "string"
+                }
+            }
+        },
+        "d3": {
+            "data": {
+                "a": [1, 2, 3, 4, 5],
+                "b": ["a", "b", "c", "d", "e"],
+                "c": [1, 2, 2, 3, 3]
+            },
+            "schemas": {
+                "SparkDFDataset": {
+                    "a": "int",
+                    "b": "string",
+                    "c": "int"
+                }
+            }
+        }
+    },
+    "tests": [
+        {
+            "func": "get_row_count",
+            "dataset": "d1",
+            "expected": 2
+        },
+        {
+            "func": "get_row_count",
+            "dataset": "d2",
+            "expected": 10
+        },
+        {
+            "func": "get_table_columns",
```

Comment:
For some reason, this is failing on pandas 0.22...any idea why?

Reply:
I looked very briefly at this, and it looked like Pandas was sorting the columns in alphabetical order on creation of the dataframe, instead of keeping the order given to it. No idea why, but I'll look into it a little more.

Reply:
Oh, I bet that's exactly it--it's probably not actually pandas, but rather that we're starting from a dictionary, which is inherently unordered.

In the wrapper code for your test construction that you use to build the datasets, you could switch to an ordereddict perhaps; I did something similar in the release 0.6.1 branch if you want to take a look at that to get python2 tests to pass for spark (I haven't merged that branch yet because of travis configuration hassles with the matrix of supported versions)

Reply:
I'll try that, but if that's the case I don't know why this wouldn't be an issue in other parts of the test suite, since we're doing a similar thing for loading datasets from json.